test/search/test_search.cpp

```diff
@@ -417,11 +417,21 @@ testRadiusSearch (typename PointCloud<PointT>::ConstPtr point_cloud, std::vector
       // compare results to each other
       #pragma omp parallel for \
         default(none) \
-        shared(distances, indices, passed, search_methods)
+        shared(distances, indices, passed, search_methods, radius)
       for (int sIdx = 1; sIdx < static_cast<int> (search_methods.size ()); ++sIdx)
       {
-        passed [sIdx] = passed [sIdx] && compareResults (indices [0],    distances [0],    search_methods [0]->getName (),
-                                                         indices [sIdx], distances [sIdx], search_methods [sIdx]->getName (), 1e-6f);
+        const bool same_results = compareResults (indices [0],    distances [0],    search_methods [0]->getName (),
+                                                  indices [sIdx], distances [sIdx], search_methods [sIdx]->getName (), 1e-6f);
+        if (!same_results) {
```

Comment:
Since we use both branches anyway, let's use the non negating form and switch the branches around?
```suggestion
        if (same_results) {
```
(don't merge this, needs more changes)